### ISSUE TITLE
Installation issue: cuda; OverflowError: signed integer is greater than maximum

### ISSUE THREAD

**Issue by ikitayama** [LLNL/spack#25141]
```
$ spack install cuda@11.3.0
[+] /p/scratch/cjzam11/kitayama1/projects/spack/opt/spack/linux-centos8-zen2/gcc-10.3.0/libiconv-1.16-zid6uv2jyijm57tvod4c3nmmzpfp7tsq
[+] /p/scratch/cjzam11/kitayama1/projects/spack/opt/spack/linux-centos8-zen2/gcc-10.3.0/pkgconf-1.7.4-kdw3p3eo4ujoumzws3emmmudx2cq7ggg
[+] /p/scratch/cjzam11/kitayama1/projects/spack/opt/spack/linux-centos8-zen2/gcc-10.3.0/xz-5.2.5-5f7lza75r4a4jjnpa7a3tykaoiqq3cxl
[+] /p/scratch/cjzam11/kitayama1/projects/spack/opt/spack/linux-centos8-zen2/gcc-10.3.0/zlib-1.2.11-ysmifax4vc6oqjhhhb7j7ozvdgfyvj4u
[+] /p/scratch/cjzam11/kitayama1/projects/spack/opt/spack/linux-centos8-zen2/gcc-10.3.0/libxml2-2.9.10-mgjltcdszna35khm4uabrv7vhxmmjbjo
==> Installing cuda-11.3.0-im3dsuofedayqbiwtir3yhz4lyqicijr
==> No binary for cuda-11.3.0-im3dsuofedayqbiwtir3yhz4lyqicijr found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/26/262da7f77db177b153a6b70b5812210f3f72f00eb608ab9cf2a4582328c4bf5c
==> Error: OverflowError: signed integer is greater than maximum

/p/scratch/cjzam11/kitayama1/projects/spack/lib/spack/spack/package.py:1378, in do_fetch:
       1375
       1376        self.stage.create()
       1377        err_msg = None if not self.manual_download else self.download_instr
  >>   1378        start_time = time.time()
       1379        self.stage.fetch(mirror_only, err_msg=err_msg)
       1380        self._fetch_time = time.time() - start_time
       1381
```

**Comment by ikitayama** [comment]
@Rombur  @ax3l would you take a look at it?

**Comment by ikitayama** [comment]
After pulling the update, I was able to build `cuda@11.4`

**Comment by ax3l** [comment]
Fixed via #24623
